community/cypher/cypher-compiler-3.1/src/main/scala/org/neo4j/cypher/internal/compiler/v3_1/planner/logical/CardinalityCostModel.scala

```diff
@@ -39,28 +39,30 @@ object CardinalityCostModel extends CostModel {
   private def costPerRow(plan: LogicalPlan): CostPerRow = plan match {
 
     case  _: NodeByLabelScan |
-         _: ProjectEndpoints
+          _: NodeIndexScan |
+          _: ProjectEndpoints
+    => FAST_STORE
+
+    // Filtering on labels and properties
+    case Selection(predicates, _) if predicates.exists {
+      case _: Property | _: HasLabels => true
+      case _ => false
+    }
     => FAST_STORE
 
     case _: AllNodesScan => 1.2
 
     case _: Expand |
-         _: VarExpand  => 2.0
+         _: VarExpand  => 0.5
```

Comment:
...or wait no thumbs up I think. The cost for the scan is already being considered

```scala
private def expandResult(graph: GraphDatabaseQueryService, scan: Pipe) = {
    val scanCost = medianPerRowCount(runSimulation(graph, scan)).head
    val simulation = runSimulation(graph, expand(scan, RELATIONSHIP)).map(_.subtractTime(scanCost.elapsed))

    simulation
  }
```